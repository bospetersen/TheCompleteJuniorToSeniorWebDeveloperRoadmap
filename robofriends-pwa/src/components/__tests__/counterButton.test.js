import Enzyme, { shallow, mount, render } from 'enzyme';
import React from 'react';
import CounterButton from '../CounterButton';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

it('Takes snapshot of CounterButton component', () => {
  const color = "red";
  expect(shallow(<CounterButton color={color} />)).toMatchSnapshot();
})

it('correctly increments counter', () => {

  const color = "red";
  const wrapper = shallow(<CounterButton color={color} />);

  wrapper.find('[id="counter"]').simulate('click')
  expect(wrapper.state()).toEqual({ count: 1 })

})