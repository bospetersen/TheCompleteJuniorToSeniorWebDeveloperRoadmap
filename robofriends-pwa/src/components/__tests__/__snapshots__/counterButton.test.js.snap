// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Takes snapshot of CounterButton component 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
